### PR TITLE
Create soft-host-to-vm-affinity-support.html.md

### PR DESCRIPTION
**Pull request by yanirq**
Changes proposed in this pull request:

- Add feature page for : Enhance affinity system to support hosts feature

I confirm that this pull request was submitted according to the [contribution guidelines](/.github/CONTRIBUTING.md): @yanirq

This pull request needs review by:  @doron-fediuck @MarSik

### CODE REVIEW

**Comment by doron-fediuck** [comment]
@bproffitt Can you tell us why it failed?

File: source/develop/release-management/features/sla/soft-host-to-vm-affinity-support.html.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+title: soft-host-to-vm-affinity-support
+category: feature
+authors: yquinn,msivak
+wiki_category: Feature
+wiki_title: Features/soft-host-to-vm-affinity-support
+wiki_last_updated: 2016-11-23
+---
```
Comment (rollandf) [inline source/develop/release-management/features/sla/soft-host-to-vm-affinity-support.html.md]: The 'wiki' tags are only for old pages. Please remove them.

Also you have missing headers in this format :
title: DiscardAfterDelete
category: feature
authors: ishaby
feature_name: Discard after delete
feature_modules: vdsm
feature_status: Partially implemented in oVirt 4.0

Comment (rollandf) [inline source/develop/release-management/features/sla/soft-host-to-vm-affinity-support.html.md]: This is failing the CI

File: source/develop/release-management/features/sla/soft-host-to-vm-affinity-support.html.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+To :      affinity_group_id (foreign key to affinity_groups + delete cascade) | vm_id (foreign key to vm_static + delete cascade) - change modifier to enable null   | vds_id (foreign key to vds_static + delete cascade) | (other resource ids in the future...)
+(any additional future resource can be added here with an index and a foreign key)
+
+* Added a constraint to check that at least one column out of vm_id and vds_id is not null for each entry.
+
+
+The table structure of affinity_groups will altered by adding a new none column:
+affinity_groups: id, name, cluster_id (foreign key to vds_groups + delete cascade), positive -> vm_positive , enforcing -> vm_enforcing ,vds_positive,vds_enforcing ,vms_affinity_enabled (true/false  - default false) 
+
+NOTE : The additional “vms_affinity_enabled”  flag is needed for affinity group to be able to express that the group of its VMs have no relationship to each other.
```
Comment (MarSik) [inline source/develop/release-management/features/sla/soft-host-to-vm-affinity-support.html.md]: Just a grammar issue: "group of its VMs have no..." should probably just say "VMs in an affinity group have no..".

**Comment by bproffitt** [comment]
(Sorry was on PTO.) Okay, looks like this is not failing anymore. Is this under general review now?

File: source/develop/release-management/features/sla/soft-host-to-vm-affinity-support.html.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    <vds_positive>true</vds_positive>
+    <vds_enforcing>true</vds_enforcing>
+    <vms_affinity_enabled>true</vms_affinity_enabled>
+  </affinity_group>
+</affinity_groups>
+```
+>**NOTE** : vm_enforcing and vm_positive tags were originally enforcing and positive.
+> in order to perserve old functionallity either : 
+> 1) the old name will be kept and represent vms. 
+> 2) vm_enforcing and vm_positive will be added and positive/enforcing tags will be deprecated
```
Comment (jhernand) [inline source/develop/release-management/features/sla/soft-host-to-vm-affinity-support.html.md]: I think that option 2) is better.

As you are introducing a set of cofigurations that are identical for virtual machines and hosts, I'd suggest to introduce a new `AffinityGroupConfiguration` type (you can probably find a better naem) grouping them. Then you can have two attributes of that type, one for virtual machines and another for hosts:

```xml
<vms_configuration>
  <positive>true|false</positive>
  <enforcing>true|false</enforcing>
  <enabled>true|false</enforcing>
</vms_configuration>
<hosts_configuration>
  <positive>true|false</positive>
  <enforcing>true|false</enforcing>
  <enabled>true|false</enforcing>
</hosts_configuration>
```

The existing `positive` and `enforcing` attributes need to be preserved, and marked as deprecated. They should have exactly the same meaning that they had before.

**Comment by bproffitt** [comment]
Is this waiting on review? Or can it be merged?

**Comment by yanirq** [comment]
@bproffitt  It can be merged .  additions/changes might be added later on.

**Pull request merged**